### PR TITLE
Issue 28983 dotcms dev latest broken

### LINKED ISSUE #28983: dotCMS-dev latest nightly exits without error
### Parent Issue

_No response_

### Problem Statement

Using [the documentation ](https://github.com/dotCMS/core/blob/master/docker/dev-env/README.md), and the [latest nightly build](https://hub.docker.com/layers/dotcms/dotcms-dev/nightly_2181d4a/images/sha256-4ec85e1fc5c2199f82e53713924e48c1f93f535f1fb0964fa81213660a33f86f?context=explore), the containers fails and exit without error.

### Steps to Reproduce

Follow the instructions [on the readme](https://github.com/dotCMS/core/blob/master/docker/dev-env/README.md), and the [latest nightly build).

TL;DR: with docker installed and running locally...

```
docker run --rm \
--pull always \
-p 8082:8082 \
-p 8443:8443 \
-v $PWD/data:/data \
-e DOTCMS_SOURCE_ENVIRONMENT=https://demo.dotcms.com/ \
-e DOTCMS_USERNAME_PASSWORD="admin@dotcms.com:admin" \
 dotcms/dotcms-dev:nightly
```

In depending on when this gets picked up we can target the specific nightly that was failing when @prestonso reported.

```
docker run --rm \
--pull always \
-p 8082:8082 \
-p 8443:8443 \
-v $PWD/data:/data \
-e DOTCMS_SOURCE_ENVIRONMENT=https://demo.dotcms.com/ \
-e DOTCMS_USERNAME_PASSWORD="admin@dotcms.com:admin" \
 dotcms/dotcms-dev:nightly_2181d4a
```

### Acceptance Criteria

1. documentation referenced in the readme should work.  dotCMS should start, run, and accept connections on https://local.dotcms.site:8443/dotAdmin should work with the dotCMS login page.
2. Should test for this in the nightly build steps 

### dotCMS Version

nightly build in Docker
dotcms/dotcms-dev:nightly_2181d4a

### Proposed Objective

Reliability

### Proposed Priority

Priority 2 - Important

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

Source: [Note in Slack](https://dotcms.slack.com/archives/C028Z3R2D/p1719236480640119)

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by sfreudenthaler**
### Proposed Changes
* update readme for dotcms-dev to reflect postgres 16 
* removes hanging `s` typo outside of quoted string in console echoing
* update /srv/entrypoint.sh to start dotCMS server when no args are provided to it
* update /srv/entrypoint.sh to use `dotcms` as the arg if no args are provided or if dotcms is the arg provided
* update dev-env /entrypoint.sh to add `dotcms` as arg when invoking /srv/entrypoint.sh 

### Checklist
- [ x ] Tests
Tested locally but not in fork
- [ N/A ] Translations
- [ X ] Security Implications Contemplated (add notes if applicable)

### Additional Notes
This doesn't include the addition of test coverage.  That will come in an additional PR once this is merged and working as expected.

### Screenshots

REFS #28983

This PR fixes: #28983

### CODE REVIEW

**Review by spbolton — APPROVED** [review]
Looks good

**Review by wezell — CHANGES_REQUESTED** [review]
@sfreudenthaler @spbolton is line 21 correct?
```
exec -- ${TOMCAT_HOME}/bin/catalina.sh run "$@"
```

Do we need to force `dotcms` there?
```
exec -- ${TOMCAT_HOME}/bin/catalina.sh run "dotcms"
```

**Comment by spbolton** [comment]
@wezell The addition of the "$@" after run was a change I made unnecessarily in the last commit.  If you check the previous one it was not there before.   As we had been setting the default CMD to "dotcms" I kept that and made it so that if you override with a different command it will run that instead.    I wanted to make sure nothing broke if "dotcms" was set as the command specifically.  even after discovering it had not had any effect as previously any command would do the same thing.   Expecting this default from the dockerfile to always be set I did not set the behavior if not CMD was assigned, forgetting that the overriding dev docker file would do this.    If you know where the "dotcms" parameter is otherwise used we should change this, otherwise setting "dotcms" or no command with this change will have the same impact of starting up the service otherwise  any other command e.g. /bin/bash or even pg_dump will run without starting the server.

**Review by victoralfaro-dotcms — APPROVED** [review]

**Review by sfreudenthaler — COMMENTED** [review]
ok @wezell and @spbolton , 

Based on the feedback I'm proposing a belt and suspenders approach.  Let me know if it's too ridged now becasue I'm not using the actual value of the passed to entrypoint if it's dotcms.

I'll update the PR description but for TL;DR
I have dev-env provide the `dotcms` argument when it calls `/srv/entrypoint.sh` and just in case /srv/entrypoint.sh ever gets given no arg, it'll use `dotcms` anyway.

File: dotCMS/src/main/docker/original/ROOT/srv/entrypoint.sh [hunk 1]
```diff
@@ -12,13 +12,13 @@ source /srv/40-custom-starter-zip.sh
...
-if [[ "$1" == "dotcms" ]]; then
+if [[ "$#" -eq 0 || "$1" == "dotcms" ]]; then
   shift
   echo ""
-  echo "Starting dotCMS ..."s
+  echo "Starting dotCMS ..."
   echo "-------------------"
   echo ""
-  exec -- ${TOMCAT_HOME}/bin/catalina.sh run "$@"
+  exec -- ${TOMCAT_HOME}/bin/catalina.sh run dotcms
```
Comment (spbolton) [inline dotCMS/src/main/docker/original/ROOT/srv/entrypoint.sh:21]: the extra "dotcms" parameter is doing nothing here.  catalina.sh only looks at the first paramter e.g. "run".  Should just remove it.

**Review by spbolton — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/docker/original/ROOT/srv/entrypoint.sh (hunk 1 above)
Comment (sfreudenthaler) [reply dotCMS/src/main/docker/original/ROOT/srv/entrypoint.sh]: 🚢

**Review by wezell — APPROVED** [review]
Good.

**Review by spbolton — APPROVED** [review]

**Pull request merged**